src/account/operations/transfer/high_level/burn_native_token.rs

```diff
@@ -0,0 +1,250 @@
+                ),
+            ];
+            self.send(outputs, options).await
+        } else {
+            unreachable!("We checked if it's an alias output before")
+        }
+    }
+
+    /// Function to burn native tokens without foundry
+    pub async fn burn_native_token_without_foundry(
+        &self,
+        native_token: (TokenId, U256),
+        options: Option<TransferOptions>,
+    ) -> crate::Result<TransferResult> {
+        log::debug!("[TRANSFER] burn_native_token_without_foundry");
+        let token_id = native_token.0;
+        let burn_token_amount = native_token.1;
+
+        let (custom_inputs, outputs) = self.select_native_token_output(token_id, burn_token_amount).await?;
+
+        let options = match options {
+            Some(mut options) => {
+                options.custom_inputs.replace(custom_inputs);
+                options.allow_burning = true;
+                Some(options)
+            }
+            None => Some(TransferOptions {
+                custom_inputs: Some(custom_inputs),
+                allow_burning: true,
+                ..Default::default()
+            }),
+        };
+
+        self.send(outputs, options).await
+    }
+
+    async fn select_native_token_output(
+        &self,
+        token_id: TokenId,
+        burn_token_amount: U256,
+    ) -> crate::Result<(Vec<OutputId>, Vec<Output>)> {
+        let account = self.read().await;
+        let mut inputs_and_outputs = Vec::new();
+        for (output_id, output_data) in account.unspent_outputs().iter() {
+            match output_data.output {
+                Output::Alias(_) | Output::Basic(_) | Output::Nft(_) => {
+                    if let Some(native_tokens) = output_data.output.native_tokens() {
+                        let mut amount = U256::from(0);
+                        let mut not_to_be_burnt_native_tokens = Vec::new();
+                        for native_token in native_tokens.iter() {
+                            if *native_token.token_id() == token_id {
+                                amount += *native_token.amount();
+                            } else {
+                                not_to_be_burnt_native_tokens.push(native_token);
+                            }
+                        }
+
+                        // If the output has a native token that we wish to burn,
+                        // clone the output but without native tokens that are to be burnt
+                        if !amount.is_zero() {
+                            let not_to_be_burnt_native_tokens = not_to_be_burnt_native_tokens.into_iter().cloned();
+
+                            let output =
+                                replace_output_native_tokens(&output_data.output, not_to_be_burnt_native_tokens)?;
+                            inputs_and_outputs.push((output_id, amount, output));
+                        }
+                    }
+                }
+                Output::Treasury(_) | Output::Foundry(_) => continue,
```

Comment:
I know it's called `burn_native_token_without_foundry`, but that's just because to differentiate between melting? If so, I would say it should still be possible to burn with a foundry instead of melting
For a native token that got minted by a foundry, it's probably in most cases desired to use the melt way instead, but a foundry could also have native tokens, that got minted by other foundries